starboard/android/shared/drm_system.cc

```diff
@@ -206,10 +211,33 @@ DrmSystem::DrmSystem(
     return;
   }
   j_media_crypto_ = env->ConvertLocalRefToGlobalRef(j_media_crypto_);
+
+  Start();
+}
+
+void DrmSystem::Run() {
+  JniEnvExt* env = JniEnvExt::Get();
+  bool result = env->CallBooleanMethodOrAbort(
+      j_media_drm_bridge_, "createMediaCryptoSession", "()Z");
+  if (result) {
+    created_media_crypto_session_.store(true);
+  }
+  if (!result && j_media_crypto_) {
+    env->DeleteGlobalRef(j_media_crypto_);
+    j_media_crypto_ = NULL;
+  }
+
+  ScopedLock scoped_lock(mutex_);
+  bool update_expected = true;
+  if (have_deferred_session_update_request_.compare_exchange_strong(
+          &update_expected, false)) {
+    deferred_session_update_request_->Generate(j_media_drm_bridge_);
```

Comment:
When the check in line 232-233 succeeds, is it possible that the main thread hits line 240 below, and continue to delete |j_media_drm_bridge_| before executing line 234?

Reply:
Good catch, `j_media_drm_bridge_` could be referenced here.